botnim/query.py

```diff
@@ -111,20 +112,97 @@ def get_index_mapping(self) -> Dict:
             logger.error(f"Failed to get index mapping: {str(e)}")
             raise
 
-def run_query(query_text: str, environment: str, bot_name: str, context_name: str, num_results: int = 7) -> List[SearchResult]:
+    def get_default_num_results(self) -> int:
+        """
+        Get the default number of results for the current context
+        
+        Returns:
+            int: Default number of results
+        """
+        # Find the specific context configuration
+        context_config = next(
```

Comment:
move this to `_load_config()` and store context_config in instance property.

Reply:
applied.